apps/evm/lib/helpers.ex

```diff
@@ -97,17 +100,15 @@ defmodule EVM.Helpers do
   def decode_signed(n) when is_nil(n), do: 0
 
   def decode_signed(n) when is_integer(n) do
-    decode_signed(:binary.encode_unsigned(n))
+    if n <= @max_uint_255 do
+      n
+    else
+      n - @max_256_ceiling
+    end
   end
 
   def decode_signed(n) when is_binary(n) do
-    <<sign::size(1), _::bitstring>> = n
-
-    if sign == 0 do
-      :binary.decode_unsigned(n)
-    else
-      :binary.decode_unsigned(n) - EVM.max_int()
```

Comment:
Do we use `EVM.max_int()` anywhere else or can that function now be removed?

Reply:
yes, it's used